### PR TITLE
fixing pythonex operators

### PR DESCRIPTION
**Pull request by jjsjann123**
## What does this PR do?

python operator module doesn't have `fmod`, but only has `mod`. Switch to use `math.fmod` instead;
python operator module doesn't have `atan2`. Switch to use `math.atan2` instead;
python operator module  doesn't have `bitwise_and`, but `and_`;
python operator module  doesn't have `bitwise_or`, but `or_`;
python operator module  doesn't have `bitwise_xor`, but `xor`;
pythonex add remainder operator

TODO:

- [x] add test. I want to wait until we enable symbolic value cache option, which makes it easier to write tests.

Follow up issue:
Investigate issue with trace on math.xxx #526

### CODE REVIEW

File: thunder/tests/test_pythonex.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import pytest
+
+import thunder
+import thunder.core
+from thunder.executors.sdpaex import sdpa_ex
+from thunder.tests.framework import requiresCUDA, run_snippet
+from thunder.tests.opinfos import get_opinfo
+from collections import namedtuple
+
+CudaVersion = namedtuple("CudaVersion", "major minor")
```
Comment (crcrpar) [inline thunder/tests/test_pythonex.py]: isn't this used?

File: thunder/tests/test_pythonex.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import pytest
+
+import thunder
+import thunder.core
+from thunder.executors.sdpaex import sdpa_ex
+from thunder.tests.framework import requiresCUDA, run_snippet
+from thunder.tests.opinfos import get_opinfo
```
Comment (crcrpar) [inline thunder/tests/test_pythonex.py]: used?

File: thunder/tests/test_pythonex.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import pytest
+
+import thunder
+import thunder.core
+from thunder.executors.sdpaex import sdpa_ex
+from thunder.tests.framework import requiresCUDA, run_snippet
```
Comment (crcrpar) [inline thunder/tests/test_pythonex.py]: used?

File: thunder/tests/test_pythonex.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import pytest
+
+import thunder
+import thunder.core
+from thunder.executors.sdpaex import sdpa_ex
```
Comment (crcrpar) [inline thunder/tests/test_pythonex.py]: used?

File: thunder/tests/test_pythonex.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import pytest
+
+import thunder
+import thunder.core
```
Comment (crcrpar) [inline thunder/tests/test_pythonex.py]: used?

File: thunder/tests/test_pythonex.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+    out_ref = ref_fn(*args)
+    assert out == out_ref
+
+
+def test_fmod():
+    def foo(a, b):
+        return a % b
+
+    _run_cache_symbolic_values(foo, foo, 2.0, 1.3)
```
Comment (crcrpar) [inline thunder/tests/test_pythonex.py:19]: Q: are these float values random or intentionally chosen?

Comment (jjsjann123) [reply thunder/tests/test_pythonex.py:19]: Just arbitrary numbers. :stuck_out_tongue_closed_eyes:

Comment (jjsjann123) [reply thunder/tests/test_pythonex.py:19]: I was hoping that I can put this in opinfo. But I needed to plumb the cache option "symbolic_values", I'm not sure if that's a knob that we'd want to add, or maybe wait until we switched to symbolic_values as default cache option?!

**Review by t-vi — APPROVED** [review]
Supergood, thank you @jjsjann123 for fixing and adding a test.

**Pull request merged**